Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -46,51 +44,49 @@ namespace CategoryTheory
 
 universe v v₁ v₂ v₃ u u₁ u₂ u₃
 
+open MonoidalCategory
+
+variable (C) in
 /--
 An instance of `ChosenFiniteProducts C` bundles an explicit choice of a binary
 product of two objects of `C`, and a terminal object in `C`.
 
 Users should use the monoidal notation: `X ⊗ Y` for the product and `𝟙_ C` for
 the terminal object.
 -/
-class ChosenFiniteProducts (C : Type u) [Category.{v} C] where
-  /-- A choice of a limit binary fan for any two objects of the category. -/
-  product : (X Y : C) → Limits.LimitCone (Limits.pair X Y)
-  /-- A choice of a terminal object. -/
-  terminal : Limits.LimitCone (Functor.empty.{0} C)
+class ChosenFiniteProducts (C : Type u) [Category.{v} C] extends MonoidalCategory C where
+  /-- The tensor unit is a terminal object. -/
+  isTerminalTensorUnit : Limits.IsTerminal (𝟙_ C)
+  /-- The first projection from the product. -/
+  fst (X Y : C) : X ⊗ Y ⟶ X
+  /-- The second projection from the product. -/
+  snd (X Y : C) : X ⊗ Y ⟶ Y
+  /-- The monoidal product is the categorical product. -/
+  tensorProductIsBinaryProduct (X Y : C) : Limits.IsLimit <| Limits.BinaryFan.mk (fst X Y) (snd X Y)
```

Comment:
Maybe `open Limits`?

Reply:
Because some name clashes with the ones in `ChosenFiniteProducts` (e.g. `prodComparison`)

Reply:
Maybe `open Limits hiding prodComparison` then?

Reply:
I might need to hide a bunch of lemmas with the prefix `prodComparison` too.

Reply:
> Because some name clashes with the ones in `ChosenFiniteProducts` (e.g. `prodComparison`)

Actually it Just Works to `open Limits`